File: src/payment/datadog.js
// Copyright The OpenTelemetry Authors
// SPDX-License-Identifier: Apache-2.0
// DataDog Migration - Payment Service Instrumentation

const tracer = require('dd-trace');

// Configure DataDog tracing
tracer.init({
  service: 'payment',
  env: process.env.DD_ENV || 'development',
  version: process.env.DD_VERSION || 'latest',
  hostname: process.env.DD_AGENT_HOST || 'datadog-agent',
  port: process.env.DD_TRACE_AGENT_PORT || 8126,
  logInjection: true,
  runtimeMetrics: true,
  profiling: true,
  appsec: false,
  plugins: {
    'http': {
      enabled: true,
      blacklist: ['/health', '/metrics']
    },
    'grpc': {
      enabled: true
    },
    'redis': {
      enabled: true
    },
    'fs': {
      enabled: false // Disable to reduce noise
    }
  },
  tags: {
    'service': 'payment',
    'env': process.env.DD_ENV || 'development',
    'version': process.env.DD_VERSION || 'latest'
  }
});

// Set global tags
tracer.setTag('service', 'payment');
tracer.setTag('env', process.env.DD_ENV || 'development');
tracer.setTag('version', process.env.DD_VERSION || 'latest');

console.log('DataDog tracing initialized for payment service');

module.exports = tracer;

File: src/product-catalog/datadog.go
// Copyright The OpenTelemetry Authors
// SPDX-License-Identifier: Apache-2.0
// DataDog Migration - Product Catalog Service Instrumentation

package main

import (
	"os"

	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/tracer"
	"gopkg.in/DataDog/dd-trace-go.v1/profiler"
)

func initDataDog() {
	// Configure DataDog tracing
	tracer.Start(
		tracer.WithService("product-catalog"),
		tracer.WithEnv(getEnv("DD_ENV", "development")),
		tracer.WithServiceVersion(getEnv("DD_VERSION", "latest")),
		tracer.WithAgentAddr(getEnv("DD_AGENT_HOST", "datadog-agent")+":"+getEnv("DD_TRACE_AGENT_PORT", "8126")),
		tracer.WithRuntimeMetrics(),
		tracer.WithAnalytics(true),
	)

	// Configure DataDog profiling
	profiler.Start(
		profiler.WithService("product-catalog"),
		profiler.WithEnv(getEnv("DD_ENV", "development")),
		profiler.WithVersion(getEnv("DD_VERSION", "latest")),
		profiler.WithAgentAddr(getEnv("DD_AGENT_HOST", "datadog-agent")+":"+getEnv("DD_TRACE_AGENT_PORT", "8126")),
		profiler.WithProfileTypes(
			profiler.CPUProfile,
			profiler.HeapProfile,
			profiler.GoroutineProfile,
		),
	)
}

func getEnv(key, defaultValue string) string {
	if value := os.Getenv(key); value != "" {
		return value
	}
	return defaultValue
}
